client/packages/system/src/Stock/DetailView/DetailView.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect } from 'react';
+import {
+  TableProvider,
+  createTableStore,
+  useTranslation,
+  DetailTabs,
+  useRowHighlight,
+  // usePluginElements,
+  // usePluginEvents,
+  // PluginEventListener,
+  useBreadcrumbs,
+  useParams,
+  useConfirmationModal,
+  useNotification,
+} from '@openmsupply-client/common';
+import { ActivityLogList } from '@openmsupply-client/system';
+import { AppBarButtons } from './AppBarButtons';
+import { useStockLine } from '../api';
+import { StockLineForm } from '../Components/StockLineForm';
+import { LedgerForm } from '../Components/Ledger';
+
+export const StockLineDetailView: React.FC = () => {
+  const { id } = useParams();
+  const {
+    query: { data, isLoading },
+    draft,
+    resetDraft,
+    updatePatch,
+    isDirty,
+    update: { update, isUpdating },
+  } = useStockLine(id);
+  const { HighlightStyles } = useRowHighlight();
+  // const { dispatchEvent, addEventListener, removeEventListener } =
+  //   usePluginEvents();
+  // const [hasChanged, setHasChanged] = useState(false);
+  const { success, error } = useNotification();
+
+  const t = useTranslation('inventory');
+  //   const navigate = useNavigate();
+  const { setSuffix } = useBreadcrumbs();
+
+  useEffect(() => {
+    if (!data) return;
+    setSuffix(data?.item.name ?? '');
+  }, [data]);
+
+  // const plugins = usePluginElements({
+  //   type: 'StockEditForm',
+  //   data: stockLine,
+  // });
+
+  const showSaveConfirmation = useConfirmationModal({
+    onConfirm: () =>
+      update()
+        .then(() => {
+          const successSnack = success(t('success.data-saved'));
+          successSnack();
+        })
+        .catch(err => {
+          const errorSnack = error(err.message);
+          errorSnack();
+        }),
```

Comment:
Not sure exactly what to display for error here.

Reply:
Do you also get two identical error snacks from this e.g. if you stop the server so network request fails? I think `GqlContext` might catch and show error snack for standard graphql errors... but we'd still need a catch block here for anything that could error before the API call, or mapping after? Not sure what the best way to do that is tbh... wonder if @mark-prins has any tips?